compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/ToolAnnotationAnalysisTask.java

```diff
@@ -0,0 +1,196 @@
+package io.ballerina.lib.np.compilerplugin;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.ModuleSymbol;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.syntax.tree.ExternalFunctionBodyNode;
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.ReturnTypeDescriptorNode;
+import io.ballerina.openapi.service.mapper.type.TypeMapper;
+import io.ballerina.openapi.service.mapper.type.TypeMapperImpl;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.swagger.v3.core.util.Json;
+import io.swagger.v3.core.util.OpenAPISchema2JsonSchema;
+import io.swagger.v3.oas.models.media.Schema;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.ballerina.lib.np.compilerplugin.Commons.findNPModule;
+import static io.ballerina.lib.np.compilerplugin.Commons.hasLlmCallAnnotation;
+import static io.ballerina.projects.util.ProjectConstants.EMPTY_STRING;
+
+public class ToolAnnotationAnalysisTask implements AnalysisTask<SyntaxNodeAnalysisContext> {
```

Comment:
Also this name is wrong. Makes sense for the agent module, but not here.

Reply:
Change the name